### ISSUE TITLE
[FEATURE REQUEST]: New WebUI Enhancements

### ISSUE THREAD

**Issue by emperorstarfinder** [Virtual-Universe/Virtual-Universe#33]
This issue is to add your thoughts and ideas for the new redesigned webui for both builtin and the standalone.  Your thoughts and ideas will help us make the experience for all who use Virtual Universe even better.

Guidelines

1. What is the idea

2. Description of what the idea will do

3. Is the feature already implemented but you are seeking an enhancement

4. How will the feature if implemented be useful to everyone

5. Other information about the idea

Please be respectful of everyone's opinions and together we will make this even better.

**Comment by LifeGold** [comment]
1. Add more complete editing profile from the site(Photo, About yourself, etc)
2. Bring the groups page(Photo, name, Creator, description, button to enter.)
3. Event(Output events + add new.) For charge free add users to the Administration.
4. In the settings to render and pay to make the change of screen name.
5. Partner(If not then no one is partner to display a link to the page of application)
The page with the following information - name of the partner, apply.
The user whose nickname you entered must come notice, and only after confirmation in the profiles displayed information(paid Service)
6. Buy regions through the site.
7. possible to implement a private messaging system on the website, as the notice in the mail does not see it.
8. Application to friends through the site. Remove with friends.
9. Conclusion friends list on profile site.
10. Conclusion friends online + region + photos.
11. Button to add to the ignore list.
12. Account funding module.

This is a brief list of what needs to be in addition to the fact that there are already on the website. For the convenience of users.

**Comment by emperorstarfinder** [comment]
Anything relating to the purchase of regions and purchase on in-world currency will be handled through the incoming AdvancedCurrency Module.  This is not something we can provide in BaseCurrency itself due to legal concerns.

**Comment by IrishwolfStarfinder** [comment]
So while I am here having some coffee (on our nice sun porch of course 
with), I reviewed LifeGold's thoughts on the new website. Most of these 
ideas seem quite plausible and are things I already am working on.

 1. Add more complete editing profile from the site(Photo, About
    yourself, etc)

This one is something that generally has to be done in-world because of 
the current way the servers work.  The best way I will implement this at 
least for the time being is to have the info pulled from the user's 
profile data they set in-world.

2. Bring the groups page(Photo, name, Creator, description, button to 
enter.)

Generally each user's web profile shows the groups they are in. I 
suppose however it probably would not be a bad idea to have a means of 
listing all groups on a grid in the webUI.  However joining the group 
will need to be done in-world.


3. Event(Output events + add new.) For charge free add users to the 
Administration.

Not a bad idea to have this.  Secondlife does have this and this 
includes a destination guide to.  I will implement.  Adding users to the 
administration of the website is done based on whether they are staff.  
Not something we will likely be changing as you don't want everyone 
having administrative capability on the site because this will cause 
problems.


4. In the settings to render and pay to make the change of screen name.

Not really sure I understand this idea.  We already have display names 
so people can call themselves what they want which they have to set 
in-world via their viewer.  Their login name now is essentially that 
their login name that they start out with in-world.  You wouldn't want 
this information to change or you end up messing up a whole kettle of 
fish with inventory and other services etc.


5. Partner(If not then no one is partner to display a link to the page 
of application) The page with the following information - name of the 
partner, apply. The user whose nickname you entered must come notice, 
and only after confirmation in the profiles displayed information(paid 
Service)

This one is essentially done in-world, however LifeGold raises a good 
and valid one here as Secondlife does this via their site. There is no 
cost to partner but a L$25 fee to dissolve the partnership.  This is all 
done via the website.  I will consult with greythane about why this is 
not already done via the webui.


6. Buy regions through site

This will be available with AdvancedCurrency but will not be available 
through BaseCurrency as real world currency is involved and for legal 
reasons we cannot directly support this in BaseCurrency.


7. Possible to implement a private messaging system on the website, as 
the notice in the mail does not see it.

Something I am willing to consider as all this would need is to act like 
a built in email service kind of like forums do now.


8. Application to friends through the site. Remove with friends.

Not Feasible.  This is something that needs to be done in-world via the 
viewer.  We want our services to be uniform here.  It is however a 
feasible idea to allow you to communicate with those on your firends 
list via the site.


9. Conclusion friends list on profile site.

This one is doable and in fact Secondlife does this one as well. Will 
implement


10. Conclusion friends online + region + photos.

The friends online would ordinarily show up on the friends list. The 
website already shows regions that are online via the regions list under 
world.  Photos the WorldView module allows you to see parcels on a 
region in 3D  It essentially takes a snapshot but greythane can explain 
that better then I could.  Having a photo gallery accessible from 
in-world to is actually a good idea worth implementing.


11. Button to add to the ignore list.

Has to be done in-world via viewer.


12. Account funding module.

This will be available with AdvancedCurrency but will not be available 
through BaseCurrency as real world currency is involved and for legal 
reasons we cannot directly support this in BaseCurrency.

These are good ideas and most I can implement in the new version.


Irishwolf Starfinder
Core Developer
Server Development and Maintenance
Work Group Member
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by emperorstarfinder** [comment]
@IrishwolfStarfinder 

 1. Add more complete editing profile from the site(Photo, About yourself, etc) 

This one is something that generally has to be done in-world because of the current way the servers work. The best way I will implement this at least for the time being is to have the info pulled from the user's profile data they set in-world.

Viewers do have the ability to do web profiles could we perhaps tap into that here to?  SL does do this to a degree but changing the profile picture etc is done in-world via viewer i believe.

**Comment by LifeGold** [comment]
Why bring all offers on this website, not everything can always be in the game, but to sit on site you can always phone, and it is a big minus that there is no there it, in fact to the user at the time when he can't be in the game and do nothing, and so it is possible though to sit and look through the group, change avatar and more, we have Russian 3D game, I played it 4 years for the past year do not come into it but the site is constantly looking can who with friends was written, etc. People need to take something, if not them something to do, and they will not move, as the sense to go somewhere in a new place if there is SL and even with 20 of 3D games, the exact same imvu, 3dx sex chat etc

**Comment by emperorstarfinder** [comment]
A lot of things we indicate might need to be done in-world via viewer are because of the way the viewers and the architecture is written.  Some things are relating to how packets are handled etc. which causes us to be limited to a degree on some things.  This doesn't mean that down the road as we make other changes to our servers etc that we wont be able to add more functionality.  When we indicate that something may not be able to be done it usually means in the short term but that it might be something we will be working on down the road.

**Comment by IrishwolfStarfinder** [comment]
EmperorStarfinder is correct here.  What I did in providing this 
information is giving the limits of what we can do in the short term.  I 
by no means am saying that these are not things we can't look at 
implementing over the long term.  In the short term we must be mindful 
of things that that create limits to ensure compatibility for all grids 
using our architecture and viewers so they wont violate policies they 
have to abide by with other grids they provide support for.

Keep in mind that the ideas posted will get fair and thorough review by 
the development team as always.  Ideas may get accepted for 
implementation in the short term while others will be accepted for 
implementation over the long term with ideas not accepted being held for 
further review at a later time.  If we reject an idea we will always say 
we have rejected the idea and the reasons why we won't accept the 
feature requested.

@EmperorStarfinder I will let you know tomorrow when I have had more 
time to think through the full menu layout as to how that will look.

Irishwolf Starfinder
Core Developer
Server Development and Maintenance
Work Group Member
Grid Architecture Development Work Group
Second Galaxy Development Team
